src/Raven.Server/Documents/Queries/FacetField.cs

```diff
@@ -65,7 +65,7 @@ public FacetOptions GetOptions(JsonOperationContext context, BlittableJsonReader
 
         public void AddAggregation(FacetAggregation aggregation, QueryFieldName name)
         {
-            Aggregations[aggregation] = name;
+            Aggregations.Add(aggregation, name);
```

Comment:
This would throw a _very_ confusing error, no?

Reply:
Yes, but the check is in QueryMetadata, not here. This is 'just in case'.

Reply:
I saw that, but if things change and we still get there, the error is going to be bad for users.
So we either revert this and use the upstream call, or do proper handling here.